app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-core.js

```diff
@@ -273,6 +273,27 @@ window.GOVUK.analyticsGa4 = window.GOVUK.analyticsGa4 || {};
               index_link: parseInt(index[2])
             }
         }
+      },
+
+      addAttributesToElements: function (targetElementsSelector, dataAttributes) {
+        var targetElements = document.querySelectorAll(targetElementsSelector)
```

Comment:
Quick question: would we want to wrap the code in this function in a guard condition to check that `targetElementsSelector` exists so that the code doesn't run when it doesn't need to? Having said that, I believe `querySelectorAll` will just return an empty `nodeList` on elements that don't exist so I don't think this code would get very far anyway.

Also sidenote nitpick: what do you think about renaming `targetElementsSelector` to just `selector`? It made it a bit more readable for me but I appreciate this very much comes down to personal preferences so happy to leave it as it is!

Reply:
Yeah, if the list is empty the for loop just won't execute so I think it should be fine to stay how it is.

I can change it to `selector` instead :+1: